### PR TITLE
nix is hard

### PR DESCRIPTION
**Pull request by isabelroses**
So um what happened here.....

- you can now build your hm config through nixos
- misc cleanup
- home-manager global pkgs means you don't need to double declare nixpkgs
- `home.file.".config/x/x"` -> `xdg.configFile"x/x"`, much nicer
- inherit is nice, don't use `with x;` without reason, `rec` sucks
- fix your ADHD occasionally
- comfysage is a much nicer name

### CODE REVIEW

File: hosts/default.nix [hunk 1]
```diff
@@ -5,13 +5,14 @@
...
-}: {
+}:
+{
   nixpkgs = {
     # You can add overlays here
     overlays = [
       # Add overlays your own flake exports (from overlays and pkgs dir):
       self.outputs.overlays.additions
-      self.outputs.overlays.modifications
+      # self.outputs.overlays.modifications
```
Comment (comfysage) [inline hosts/default.nix]: why this?

File: lib/builders.nix [hunk 2]
```diff
@@ -7,22 +7,20 @@
   inherit (inputs) self;
 
   perInput = system: flake:
-    lib.optionalAttrs (flake ? formatter.${system}) {
-      formatter = flake.formatter.${system};
-    } // lib.optionalAttrs (flake ? packages.${system}) {
-      packages = flake.packages.${system};
-    };
+    lib.optionalAttrs (flake ? formatter.${system}) {formatter = flake.formatter.${system};}
+    // lib.optionalAttrs (flake ? packages.${system}) {packages = flake.packages.${system};};
```
Comment (comfysage) [inline lib/builders.nix]: i actually prefer the original formatting here

File: hosts/default.nix (hunk 1 above)
Comment (isabelroses) [reply hosts/default.nix]: I commented the over modifications out since it wasn't doinging anything

File: lib/builders.nix (hunk 2 above)
Comment (isabelroses) [reply lib/builders.nix]: nix formatting is cursed

File: overlays/default.nix [hunk 3]
```diff
@@ -5,23 +5,22 @@
...
-    # example = prev.example.overrideAttrs (oldAttrs: rec {
-    # ...
-    # });
-  };
+  # https://wiki.nixos.org/wiki/Overlays
+  # modifications = final: prev: {
+  #   example = prev.example.overrideAttrs (oldAttrs: rec {
+  #   ...
+  #   });
+  # };
```
Comment (comfysage) [inline overlays/default.nix]: i prefer to keep this uncommented. idk why

Comment (isabelroses) [reply overlays/default.nix]: you so silly :3

**Comment by comfysage** [comment]
fix with #4

**Pull request closed without merging**